add s390x support for kubevirt builder.

vamsikrishna-siddu opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:
This pr adds the s390x support to kubevirt builder image.
The kubevirt builder image for s390x does not contain Bazel. We are not including the Bazel because at the moment there is no upstream official support of Bazel for s390x. We are working with Bazel community to get the support. But it will take lot of time to get the support as the Bazel community has other priorities and they are not willing to add the support now. We also thought to publish the natively compiled Bazel Binary for s390x, We were not able to do it because of legal issues. We will include the Bazel in the  kubevirt builder once we get the Bazel support on s390x.
**How we are proceeding for unit and E2E Tests**.
Unit Test:
We will run the unit-tests using Golang . We will publish the kubevirt builder image without Bazel to the public and use that image to run the unit tests using make go-test.

E2E Test:
We will use our own kubevirt builder image (which contains Bazel) from our internal registry, and we will override the used image using environment variable to run the e2e tests.

cc: @brianmcarey 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
add s390x support for kubevirt builder
```

alicefr commented on hack/builder/Dockerfile:
```diff
@@ -130,6 +142,20 @@ COPY entrypoint.sh /entrypoint.sh
+
+RUN  mkdir -p /usr/s390x-bazel
+
+COPY ./bazel /usr/s390x-bazel/
```
Doesn't this command need to be run only on s390x?I suppose we don't want to copy the binary also on the arm and x86 image. However, I'm not sure how you can conditionally run the copy commad in a Dockerfile

alicefr replied:
Additionally, where is this binary fetched locally?

alicefr replied:
It is a bit convoluted but I like the suggestion from here for a real conditional copy: https://stackoverflow.com/questions/31528384/conditional-copy-add-in-dockerfile

You could create 2 intermediate building steps with the command you used below for fetching bazel and then copy the binary from the intermediate step. I hope it is clear

ibm-jitendra replied:
@vamsikrishna-siddu 

Please put appropriate condition for s390x arch.

jschintag replied:
I think either making it a multistage build, or alternatively putting the s390x build into a separate Dockerfile are both options to consider. Especially since the long term strategy would be to have bazel on s390x, making this a hopefully temporary solution.
When splitting it into a separate Dockerfile and/or multistage build, it might also be good to include the building bazel part as a stage. This could be easily done using https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh (needs an ubuntu stage). That way the actual logic for building bazel is not maintained inside the kubevirt project, we would just be fetching and adapting an existing build script.

alicefr replied:
Please, try to minimize the code changes. So, I would avoid a second Dockerfile, I would be in favor of multi stage building. Something like:
```dockerfile
ARG BUILD_ARCH

FROM <build-bazel>:s390x as bazel-s390x
# Fetch here the bazel versio 


FROM quay.io/centos/centos:stream9 as bazel-default
curl -L -o /usr/bin/bazel https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-linux-${BAZEL_ARCH} && chmod u+x /usr/bin/bazel;

# In the last stage
COPY --from=bazel-${BUILD_ARCH} /usr/bin/bazel /usr/bin/bazel
```

Then, we can poss `--build-arg=s390x` or `--build-arg=default` and select the stage based on the arch

jschintag replied:
I think it meight be better to put the condition logic into a single bazel stage, and just do
```
COPY --from=bazel /bazel /usr/bin/bazel
```
instead. That way we can just use the normal arch arg and don't need a new one.

alicefr replied:
This work too if you already have the bazel binary ready somewhere to fetch. Is this the case?

jschintag replied:
Unfortunately not, but i was thinking more about 2 separate `RUN` steps, one if arch is s390x, one if not.

vamsikrishna-siddu replied:
> Additionally, where is this binary fetched locally?
@alicefr  I have included the bazel at this place hack/builder/bazel 
https://github.com/kubevirt/kubevirt/pull/11097/files#diff-a3062a93f41b73857652d6d8edacd4b84e21c40ddca7b74189bc7845ab0e630a

vamsikrishna-siddu replied:
> It is a bit convoluted but I like the suggestion from here for a real conditional copy (second answer): https://stackoverflow.com/questions/31528384/conditional-copy-add-in-dockerfile
> 
> You could create 2 intermediate building steps with the command you used below for fetching bazel and then copy the binary from the intermediate step based on the ARCH. I hope it is clear

Hi @alicefr  I have followed your suggestion and implemented a conditional-copy-add that will copy the bazel binary if the architecture is s390x else it will fetch the binary from the github releases.
you can see my latest commit.
https://github.com/kubevirt/kubevirt/pull/11097/commits/8817df01bca5819b69a23238f634ea09a3937632

alicefr commented on hack/builder/Dockerfile:
```diff
@@ -130,6 +142,20 @@ COPY entrypoint.sh /entrypoint.sh
+#Bazel support for s390x is not available in upstream
+#Copy the pre-built bazel-s390x binary to builder image
+
+RUN  mkdir -p /usr/s390x-bazel
```
Same as below, don't we need this directory only on s390x?

jschintag commented on hack/builder/publish.sh:
```diff
@@ -29,5 +29,9 @@ for ARCH in ${ARCHITECTURES}; do
+# The below commands are not working with podman container runtime.
+# Using docker container runtime to create and push the manifest.
+
+export KUBEVIRT_CRI=docker
```
Where does this line come from and is it needed?

vamsikrishna-siddu replied:
@jschintag  The command to create and push the manifest are not working with any other container runtimes execpt docker. So i am setting the KUBEVIRT_CRI to docker

brianmcarey replied:
@vamsikrishna-siddu - which version of podman did you try this with? It should be working on some of the later versions of podman

vamsikrishna-siddu replied:
@brianmcarey 
```
Version:      3.4.2
API Version:  3.4.2
Go Version:   go1.15.2
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

brianmcarey replied:
I think it works in podman version > 4.4 

Anyway this script is used by a CI job to publish the builder images so updates here should not be required.

vamsikrishna-siddu replied:
Sure @brianmcarey  I verified in 4.4.1 version it is working. Let me remove it. Thanks

fabiand commented:
Howdy.

Please include an `OWNERS` file to specify who is owning s390 support.

Setting a

/hold

until the file is added.

brianmcarey commented:
> Howdy.
> 
> Please include an `OWNERS` file to specify who is owning s390 support.
> 
> Setting a
> 
> /hold
> 
> until the file is added.
> 
> For example: Add a new member to OWNERS in #11046 with a comment that they/she/he is covering for s390.

I think this is something we will have to do retroactively as these are some of the first s390x contributions - the contributors are not yet part of the kubevirt org and can't be added to any special interest groups as a result.

What do you think? 

> 
> @dhiller @brianmcarey can help to do this modification

alicefr commented on hack/builder/Dockerfile:
```diff
@@ -1,5 +1,18 @@
+FROM quay.io/centos/centos:stream9 as build_s390x
+#Bazel support for s390x is not available in upstream
+#Copy the pre-built bazel-s390x binary to builder image
+ONBUILD COPY ./bazel /usr/bin/bazel
```
I agree with @brianmcarey. However, I would avoid to do the build as part of KubeVirt
Would be possible that you publish a container image containing the bazel binary for s390x?
It doesn't necessary need to be an image, a release somewhere can be also another option, but with an image you don't need an extra repository and you can directly reference it.

alicefr replied:
@vamsikrishna-siddu I'm not sure this copy is correct. Probably the `ONBUILD` suggestion was wrong from my part. However, this small example does the trick:
```dockerfile
ARG stage=B
ARG build_stage_var=STAGE${stage}

FROM alpine as STAGEA
RUN echo "a" > /test

FROM centos as STAGEB
RUN echo "b" > /test

FROM $build_stage_var as BUILD

FROM centos
COPY --from=BUILD /test /
```
You can try with:
+ `$ podman build --build-arg "stage=A" -t test  .` -> it creates the file with a
*  `$ podman build --build-arg "stage=B" -t test  .`  -> it creates the file with b
*  `$ podman build " -t test  .`  -> it creates the file with b as it is the default

alicefr replied:
What I like of this approch  is that if the stage a isn't selected, then alpine isn't pulled and the RUN command in STAGEA isn't run

vamsikrishna-siddu replied:
Hi @alicefr What i understand when you said publish a container image containing the bazel binary for s390x is that we will use some alpine base image and put the s390x bazel binary inside it and we publish that image to dockerhub and we use that image in the above dockerfile and copy the bazel binary from it. am i correct here?

vamsikrishna-siddu replied:
Actually iam currently working on to build the bazel during the kubevirt build by using this script which will build the bazel for s390x  https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh .

vamsikrishna-siddu replied:
> I agree with @brianmcarey. However, I would avoid to do the build as part of KubeVirt Would be possible that you publish a container image containing the bazel binary for s390x? It doesn't necessary need to be an image, a release somewhere can be also another option, but with an image you don't need an extra repository and you can directly reference it here.

Regarding this i will discuss with my team and see what i can do here. Thanks

alicefr replied:
> Actually iam currently working on to build the bazel during the kubevirt build by using this script which will build the bazel for s390x https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh .

I would strongly discourage to build bazel as part of KubeVirt builds. Building bazel takes quite a while, it consumes KubeVirt CI resources and if something goes wrong, then it might slow down or block KubeVirt CI.

I think this needs to be something separate and either publish officially by the bazel community or something well supported by IBM.

alicefr replied:
> Hi @alicefr What i understand when you said publish a container image containing the bazel binary for s390x is that we will use some alpine base image and put the s390x bazel binary inside it and we publish that image to dockerhub and we use that image in the above dockerfile and copy the bazel binary from it. am i correct here?

As I mentioned above, it doesn't necessary need to be a container image. But I think here, it will be easier to reference, but it was only a suggestion.

Again, it doesn't need to be either a container image or alpine. The decision how you publish the bazel binary is up to you. If this is done only for kubeVirt, you can use whatever base image you want, even an empty one.

vamsikrishna-siddu commented:
Hello @brianmcarey @alicefr  As the Bazel support is not present for s390x. We have the below proposal for enabling the kubevirt on s390x. We want to know from your side whether we can go ahead with this proposal?

**Unit Test** 
We will run the unit-tests using Golang until we get the Bazel support on s390x. We will publish the kubevirt builder image without Bazel to the public and use that image to run the unit tests using make go-test. 

**E2E Test**
At present we can't publish Bazel binary for s390x. We will use our own kubevirt builder image (which contains Bazel) from our internal registry, and we will override the used image using environment variable to run the e2e tests.

brianmcarey commented:
> Hello @brianmcarey @alicefr As the Bazel support is not present for s390x. We have the below proposal for enabling the kubevirt on s390x. We want to know from your side whether we can go ahead with this proposal?

Hi @vamsikrishna-siddu 
> 
> **Unit Test** We will run the unit-tests using Golang until we get the Bazel support on s390x. We will publish the kubevirt builder image without Bazel to the public and use that image to run the unit tests using make go-test.
> 
> **E2E Test** At present we can't publish Bazel binary for s390x. We will use our own kubevirt builder image (which contains Bazel) from our internal registry, and we will override the used image using environment variable to run the e2e tests.

Would you be able to publish this builder image to quay.io ? In this case we could just use this image for both the unit test and e2e tests - this would satisfy CI. It would have to be documented that contributors wishing to build or test against s390x would have to use this builder image. 

@dhiller @xpivarc - what do you think of this ?

dhiller commented:
> > Hello @brianmcarey @alicefr As the Bazel support is not present for s390x. We have the below proposal for enabling the kubevirt on s390x. We want to know from your side whether we can go ahead with this proposal?
> 
> Hi @vamsikrishna-siddu
> 
> > **Unit Test** We will run the unit-tests using Golang until we get the Bazel support on s390x. We will publish the kubevirt builder image without Bazel to the public and use that image to run the unit tests using make go-test.
> > **E2E Test** At present we can't publish Bazel binary for s390x. We will use our own kubevirt builder image (which contains Bazel) from our internal registry, and we will override the used image using environment variable to run the e2e tests.
> 
> Would you be able to publish this builder image to quay.io ? In this case we could just use this image for both the unit test and e2e tests - this would satisfy CI. It would have to be documented that contributors wishing to build or test against s390x would have to use this builder image.
> 
> @dhiller @xpivarc - what do you think of this ?

Sounds good to me.

dhiller commented:
> > Howdy.
> > Please include an `OWNERS` file to specify who is owning s390 support.
> > Setting a
> > /hold
> > until the file is added.
> > For example: Add a new member to OWNERS in #11046 with a comment that they/she/he is covering for s390.
> 
> I think this is something we will have to do retroactively as these are some of the first s390x contributions - the contributors are not yet members of the kubevirt org and can't be added to any special interest groups as a result.
> 
> What do you think?
> 
> > @dhiller @brianmcarey can help to do this modification

I think we might need to add some special SIGs covering the arch support, like i.e. `sig-arm` or `sig-s390x`. There we can add ourselves (me and Brian) on an interim basis and then add people that are responsible for this and will support this. WDYT?

brianmcarey commented:
> and Brian) on an interim basis and then add people that are responsible for this and will support this.

This sounds like a good idea.

brianmcarey commented on hack/bazel-server.sh:
```diff
@@ -1,5 +1,16 @@
+  sleep 1
+else
+  #Sleep for infinite time to run the s390x unit tests.
+   while true; do
```
Why is this infinite sleep required?

jschintag replied:
The script is called as the run command for the builder container. As such it is the "main" process, whereas the other steps are executed via `exec`. Therefore the container exits when the `hack/bazel-server.sh` script exits. An alternative to infinite sleep we could maybe do something like 4h or another sensible timeout?

Regardles, @vamsikrishna-siddu i would not use an endless loop here but rather call `sleep infinity` directly, to avoid wasting CPU cycles on the while loop.

vamsikrishna-siddu replied:
Hi @jschintag  i removed the endless loop and changed the command to sleep infinity

brianmcarey commented on hack/builder/Dockerfile:
```diff
@@ -130,10 +141,13 @@ COPY entrypoint.sh /entrypoint.sh
 RUN mkdir -p /root/go/src/kubevirt.io/kubevirt && git config --global --add safe.directory /root/go/src/kubevirt.io/kubevirt
 
 ENTRYPOINT [ "/entrypoint.sh" ]
+
```
nit - no need for this new line here

jschintag replied:
I think adding newlines at the end of a file if there is none is the default setting on most modern editors.
So they might not be required, but are mostly enforced by editors by default...

vamsikrishna-siddu replied:
removed the new line.

brianmcarey reviewed (commented):
Thanks @vamsikrishna-siddu - just a couple of comments and questions inline.

vamsikrishna-siddu commented on hack/bazel-server.sh:
```diff
@@ -1,5 +1,16 @@
+ARCH=$(uname -m)
+
+# s390x builder image is not shipped with bazel.
+# Below code is causing the unit test failure in s390x.
```
done

vamsikrishna-siddu commented:
> Thanks @vamsikrishna-siddu - just a couple of comments and questions inline.

Hi @brianmcarey  I have addressed your review comments and questions. Can you please review the PR again. Thanks

jschintag commented on hack/bazel-server.sh:
```diff
@@ -1,5 +1,13 @@
+ARCH=$(uname -m)
+
+# s390x builder image is not shipped with bazel.
+if [ $ARCH != "s390x" ]; then
```
@vamsikrishna-siddu Looking at this part again, i think we need to change this to using an injected environment variable or a cli argument instead, since otherwise this script will not run on s390x when we are triggering e2e tests.
Since we are planning on using a non-public image with bazel included in the future.
Alternatively we could also make the condition check for if bazel binary is present, though that might be instable, for example when using bazelisk...

vamsikrishna-siddu replied:
Hi @jschintag  sure will do that

vamsikrishna-siddu replied:
Hi @jschintag  I am using an environment variable called KUBEVIRT_RUN_UNIT_TEST_NO_BAZEL and injecting into the docker container. if this variable KUBEVIRT_RUN_UNIT_TEST_NO_BAZEL is set to true. Then it will sleep for infinite time otherwise it will invoke the bazel in the bazel-server.sh. Let me know your thoughts on this?

jschintag replied:
I looked at the code again and there is already an existing env variable given to the `hack/dockerized` script called `KUBEVIRT_NO_BAZEL`. It is also already set to true when invoking `make go-test`. See https://github.com/kubevirt/kubevirt/blob/main/Makefile#L79-L80.
I would like to use the existing variable instead of introducing a new, duplicated one.

vamsikrishna-siddu replied:
Hi @jschintag  I have done the changes to use KUBEVIRT_NO_BAZEL. Please review it again Thanks

jschintag replied:
Thank you, looks good

vamsikrishna-siddu commented on Makefile:
```diff
@@ -62,7 +62,7 @@ client-python:
 
 go-build:
-	hack/dockerized "export KUBEVIRT_NO_BAZEL=true && KUBEVIRT_VERSION=${KUBEVIRT_VERSION} KUBEVIRT_GO_BUILD_TAGS=${KUBEVIRT_GO_BUILD_TAGS} KUBEVIRT_RELEASE=${KUBEVIRT_RELEASE} ./hack/build-go.sh install ${WHAT}" && ./hack/build-copy-artifacts.sh ${WHAT}
+	KUBEVIRT_NO_BAZEL=true hack/dockerized "export KUBEVIRT_VERSION=${KUBEVIRT_VERSION} && KUBEVIRT_GO_BUILD_TAGS=${KUBEVIRT_GO_BUILD_TAGS} KUBEVIRT_RELEASE=${KUBEVIRT_RELEASE} ./hack/build-go.sh install ${WHAT}" && ./hack/build-copy-artifacts.sh ${WHAT}
```
yes

vamsikrishna-siddu commented on hack/dockerized:
```diff
@@ -14,6 +14,7 @@ fail_if_cri_bin_missing
 
 KUBEVIRT_BUILDER_IMAGE=${KUBEVIRT_BUILDER_IMAGE:-"quay.io/kubevirt/builder:2312220935-5cd924bf8"}
 
+
```
removed the new line.

vamsikrishna-siddu commented:
> Looks good @vamsikrishna-siddu - just the last couple of comments. And could you squash the commits addressing review comments?

I Squashed all the commits addressing review comments into a single commit. Can you please review it again. Thanks

brianmcarey reviewed (commented):
/lgtm

Thanks @vamsikrishna-siddu  - can you just add an explanation to the PR description why you are not including bazel in the s390x version of the builder image?

brianmcarey reviewed (commented):
Actually looking at the commits again - there is further more tidying up required - the changes in this PR could be handled in a single commit. 

For instance - the following commits are no longer relevant: 
- https://github.com/kubevirt/kubevirt/pull/11097/commits/19635052fd8d316be7d88be93f35b89b8486104a
- https://github.com/kubevirt/kubevirt/pull/11097/commits/947bf6b0fe3c5a103474b6d253828d3d37672aa2

/lgtm cancel

vamsikrishna-siddu commented:
> Actually looking at the commits again - there is further more tidying up required - the changes in this PR could be handled in a single commit.
> 
> For instance - the following commits are no longer relevant:
> 
> * [1963505](https://github.com/kubevirt/kubevirt/commit/19635052fd8d316be7d88be93f35b89b8486104a)
> * [947bf6b](https://github.com/kubevirt/kubevirt/commit/947bf6b0fe3c5a103474b6d253828d3d37672aa2)
> 
> /lgtm cancel

Hi @brianmcarey  I put all the changes into a single commit and updated the PR. Please review it again. Thanks

brianmcarey commented on hack/builder/Dockerfile:
```diff
@@ -130,8 +131,10 @@ COPY entrypoint.sh /entrypoint.sh
-    chmod u+x /usr/bin/bazel
+RUN if test "${ARCH}" != "s390x"; then \
+       curl -L -o /usr/bin/bazel https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-linux-${BAZEL_ARCH} && \
+    chmod u+x /usr/bin/bazel; \
```
Can you fix the indentation here?

vamsikrishna-siddu replied:
@brianmcarey done

brianmcarey reviewed (commented):
/lgtm

thanks @vamsikrishna-siddu

dhiller commented:
> > Howdy.
> > Please include an `OWNERS` file to specify who is owning s390 support.
> > Setting a
> > /hold
> > until the file is added.
> > For example: Add a new member to OWNERS in #11046 with a comment that they/she/he is covering for s390.
> > @dhiller @brianmcarey can help to do this modification
> 
> @vamsikrishna-siddu the requested change wrt ownership is still missing. I know that this is currently not possible since you are not (yet) a member of the org, but someone in `sig-buildsystem` needs to own this.
> 
> So I would suggest you apply for KubeVirt membership (see [here](https://github.com/kubevirt/community/blob/main/membership_policy.md?rgh-link-date=2024-03-15T13%3A15%3A04Z#member) how to do this), and then we can add you to the org.
> 
> I believe we should make an exception for this case wrt/ contributions, and I think this is fair, since addition of a new build architecture is significant enough.
> 
> WDYT @fabiand @vladikr @davidvossel ?

WRT sponsorship you can add @dhiller since I'm happy to sponsor it . maybe @brianmcarey also, I guess?

vamsikrishna-siddu commented:
> > Howdy.
> > Please include an `OWNERS` file to specify who is owning s390 support.
> > Setting a
> > /hold
> > until the file is added.
> > For example: Add a new member to OWNERS in #11046 with a comment that they/she/he is covering for s390.
> > @dhiller @brianmcarey can help to do this modification
> 
> @vamsikrishna-siddu the requested change wrt ownership is still missing. I know that this is currently not possible since you are not (yet) a member of the org, but someone in `sig-buildsystem` needs to own this.
> 
> So I would suggest you apply for KubeVirt membership (see [here](https://github.com/kubevirt/community/blob/main/membership_policy.md#member) how to do this), and then we can add you to the org.
> 
> I believe we should make an exception for this case wrt/ contributions, and I think this is fair, since addition of a new build architecture is significant enough.
> 
> WDYT @fabiand @vladikr @davidvossel ?

Hi @dhiller  I am  successfully now a member of kubevirt organization. can you please add me to the required special interest group. Thanks

dhiller commented:
> > > Howdy.
> > > Please include an `OWNERS` file to specify who is owning s390 support.
> > > Setting a
> > > /hold
> > > until the file is added.
> > > For example: Add a new member to OWNERS in #11046 with a comment that they/she/he is covering for s390.
> > > @dhiller @brianmcarey can help to do this modification
> > 
> > 
> > @vamsikrishna-siddu the requested change wrt ownership is still missing. I know that this is currently not possible since you are not (yet) a member of the org, but someone in `sig-buildsystem` needs to own this.
> > So I would suggest you apply for KubeVirt membership (see [here](https://github.com/kubevirt/community/blob/main/membership_policy.md?rgh-link-date=2024-03-27T13%3A28%3A10Z#member) how to do this), and then we can add you to the org.
> > I believe we should make an exception for this case wrt/ contributions, and I think this is fair, since addition of a new build architecture is significant enough.
> > WDYT @fabiand @vladikr @davidvossel ?
> 
> Hi @dhiller I am successfully now a member of kubevirt organization. can you please add me to the required special interest group. Thanks

Hey @vamsikrishna-siddu I'd suggest you create a PR against the OWNERS_ALIASES file in kubevirt/kubevirt, adding you to the [`sig-buildsystem-reviewers`](https://github.com/kubevirt/kubevirt/blob/main/OWNERS_ALIASES#L139) group, maybe with a comment after your name to make clear what you are owning.

dhiller commented:
/release-note-edit ```add s390x support for kubevirt builder```

vamsikrishna-siddu commented:
Hi @fabiand @dhiller ,
 with the recent communication in google-groups regarding this PR https://github.com/kubevirt/kubevirt/pull/11614 . I got that you are planning to implement working-groups to specify the stakeholders for the respective arches. I assume it will take some time to get that feature. meanwhile is it possible to merge this Kubevirt builder pr as it helps us to enable CI and kubevirt on s390x. Once the working-groups feature is out. We can add the contributors there.

fabiand commented:
@vamsikrishna-siddu yes, working groups.

It's not a technical feature, it's more about defining ownerhsip, and specific owners, for archtiectures.

Please join the call to speed up this process.

brianmcarey commented:
/ok-to-test

brianmcarey commented:
The arch based WG formation is still being worked on[1] but I can't see why it should hold up this PR - we need the builder image published to enable some s390x testing.

/hold cancel

[1] https://github.com/kubevirt/community/pull/298

brianmcarey commented:
/retest-required

vamsikrishna-siddu commented:
/test all

brianmcarey reviewed (commented):
/lgtm

thanks @vamsikrishna-siddu

alicefr commented:
/approve
looks good
